File: BeatPlannerAndroid/src/beatplanner/core/Track.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.IO;

namespace BeatPlanner
{


  public class Track
  {
    private readonly List<Sequence> sequences = new List<Sequence>();

    public Sequence[] Sequences
    {
      get
      {
        Sequence[] result = new Sequence[sequences.Count];
        sequences.CopyTo(result);
        return result;
      }
    }

    public void AppendSequence(Beat beat, int reps)
    {
      sequences.Add(new Sequence(beat, reps));
    }

    public void PrependSequence(Beat beat, int reps)
    {
      sequences.Insert(0, new Sequence(beat, reps));
    }

    public void InsertSequence(Beat Beat, int reps, int i)
    {
      sequences.Insert(i, new Sequence(Beat, reps));
    }
    // METER BPM REPS
    // 4/4   180 4
    private static Sequence ParseSequence(string line)
    {
      string[] splits = line.Split(new []{ ' ' }, StringSplitOptions.RemoveEmptyEntries);
      Beat beat = Beat.Parse(splits);
      int reps = Convert.ToInt32(splits[2]);
      return new Sequence(beat, reps);
    }

    public static Track Parse(string[] lines)
    {
      Track track = new Track();
      Sequence seq;
      for (int i = 0; i < lines.Length; i++)
      {
        seq = ParseSequence(lines[i]);
        track.AppendSequence(seq.Beat, seq.Reps);
      }
      return track;
    }

    public static Track Parse(string input)
    {
      return Parse(input.Split(new []{ '\r', '\n' }, StringSplitOptions.RemoveEmptyEntries));
    }

    public string Compose()
    {
      StringBuilder sb = new StringBuilder();
      foreach (Sequence seq in sequences)
      {
        sb.AppendFormat("{0} {1} {2}{3}", seq.Beat.Meter, seq.Beat.BPM, 
          seq.Reps, Environment.NewLine);
      }
      return sb.ToString();
    }

    public void Save(string path)
    {
      File.WriteAllText(path, Compose());
    }

    public static Track Load(string path)
    {
      string[] lines = File.ReadAllLines(path);
      return Parse(lines);
    }
  }

}

File: app/src/main/java/dk/aschoen/beatplanner/todo/Track.java
//
// Translated by CS2J (http://www.cs2j.com): 2014-07-29 22:02:12
//

package dk.aschoen.beatplanner.todo;


import dk.aschoen.beatplanner.core.Beat;
import dk.aschoen.beatplanner.core.Sequence;
//
//public class activity_track
//{
//    private final List<Sequence> sequences = new List<Sequence>();
//    public Sequence[] getSequences() throws Exception {
//        Sequence[] result = new Sequence[sequences.Count];
//        sequences.CopyTo(result);
//        return result;
//    }
//
//    public void appendSequence(Beat beat, int reps) throws Exception {
//        sequences.Add(new Sequence(beat,reps));
//    }
//
//    public void prependSequence(Beat beat, int reps) throws Exception {
//        sequences.Insert(0, new Sequence(beat,reps));
//    }
//
//    public void insertSequence(Beat Beat, int reps, int i) throws Exception {
//        sequences.Insert(i, new Sequence(Beat,reps));
//    }
//
//    // METER BPM REPS
//    // 4/4   180 4
//    private static Sequence parseSequence(String line) throws Exception {
//        String[] splits = line.Split(StringSplitOptions.RemoveEmptyEntries);
//        Beat beat = Beat.Parse(splits);
//        int reps = Integer.valueOf(splits[2]);
//        return new Sequence(beat,reps);
//    }
//
//    public static activity_track parse(String[] lines) throws Exception {
//        activity_track track = new activity_track();
//        Sequence seq;
//        for (int i = 0;i < lines.Length;i++)
//        {
//            seq = ParseSequence(lines[i]);
//            track.appendSequence(seq.beat,seq.reps);
//        }
//        return track;
//    }
//
//    public static activity_track parse(String input) throws Exception {
//        return Parse(input.Split(StringSplitOptions.RemoveEmptyEntries));
//    }
//
//    public String compose() throws Exception {
//        StringBuilder sb = new StringBuilder();
//        for (Object __dummyForeachVar0 : sequences)
//        {
//            Sequence seq = (Sequence)__dummyForeachVar0;
//            sb.AppendFormat("{0} {1} {2}{3}", seq.beat.Meter, seq.beat.BPM, seq.reps, Environment.NewLine);
//        }
//        return sb.ToString();
//    }
//
//    public void save(String path) throws Exception {
//        File.WriteAllText(path, compose());
//    }
//
//    public static activity_track load(String path) throws Exception {
//        String[] lines = File.ReadAllLines(path);
//        return Parse(lines);
//    }
//
//}


